pkg/networkserver/observability.go

```diff
@@ -239,11 +238,7 @@ func registerReceiveUplink(ctx context.Context, msg *ttnpb.UplinkMessage) {
 }
 
 func registerMergeMetadata(ctx context.Context, msg *ttnpb.UplinkMessage) {
-	gtws := make(map[string]struct{}, len(msg.RxMetadata))
-	for _, md := range msg.RxMetadata {
-		gtws[unique.ID(ctx, md.GatewayIdentifiers)] = struct{}{}
-	}
-	nsMetrics.uplinkGateways.WithLabelValues(ctx).Observe(float64(len(gtws)))
+	nsMetrics.uplinkGateways.WithLabelValues(ctx).Observe(float64(len(msg.RxMetadata)))
```

Comment:
With this change, the metric can no longer be called `uplinkGateways`. At least for non-packetbroker metadatas it now indicates the number of antennas, not the number of gateways that received the uplink.